java/vector/src/main/java/org/apache/arrow/vector/VectorSchemaRoot.java

```diff
@@ -54,7 +54,7 @@
   private Schema schema;
   private int rowCount;
   private final List<FieldVector> fieldVectors;
-  private final Map<String, FieldVector> fieldVectorsMap = new HashMap<>();
+  private final Map<Field, FieldVector> fieldVectorsMap = new LinkedHashMap<>();
```

Comment:
is there a reason no to change this to a Map<String, List<FieldVector>>?

Reply:
I didn't want to lose the type information. This still allows you to address Field("x", int) compared to Field("x", long).